service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -490,6 +490,20 @@ public boolean updateResource(
     return updateResourceWorker(workspaceId, resourceId, name, description, /*attributes=*/ null);
   }
 
+  @WriteTransaction
+  public boolean updateResourceCloningInstructions(
```

Comment:
Why do we need this method? I think we should generalize the `updateResource()` method. We have been making incremental changes. Perhaps we need a more general way to say: here is a resource object, please update these fields. Maybe?

In any case, we should update the metadata in one transaction.

Reply:
What you're describing is starting to sound like a data transfer object, and you can use Spring to build repositories around such objects that give you many kinds of queries for "free". I could do this a couple of ways. One way is to pass in a map of key-value pairs which gets added to the params map. Another would be to walk all the fields of the resource structure and update all the non-null values, though in that case we'd frequently be updating values that didn't change, which is theoretically harmless, if unapetizing.